include/caffe/filler.hpp

```diff
@@ -157,6 +157,27 @@ class XavierFiller : public Filler<Dtype> {
   }
 };
 
+/**
+ * @brief Fills a Blob with values @f$ x \sim N(-a, +a) @f$ where @f$ a @f$ is
+ *        set by the number of incoming nodes, based on the paper [He,
+ *        Zhang, Ren and Sun 2015]
+ */
+template <typename Dtype>
+class MSRFiller : public Filler<Dtype> {
+ public:
+  explicit MSRFiller(const FillerParameter& param)
+      : Filler<Dtype>(param) {}
+  virtual void Fill(Blob<Dtype>* blob) {
+    CHECK(blob->count());
+    int fan_in = blob->count() / blob->num();
```

Comment:
In my understanding, they use number of output channels instead of input in order to avoid increasing/decreasing gradients though backward pass, don't they? In this case, it should be the following.

    fan_in = blob->count() / blobs->channels()

I am sorry if my understanding is not correct.